pkg/controllers/ingress/ingress.go

```diff
@@ -326,12 +326,12 @@ func (c *Controller) ensureIngress(ingress *networkingv1.Ingress, ingressClass *
 	}
 
 	backendSetsToCreate := desiredBackendSets.Difference(actualBackendSets)
-
+	extractCaFromTlsCrt := util.GetIngressExtractCaFromTlsCrt(ingress)
 	for _, bsName := range backendSetsToCreate.List() {
 		startBuildTime := util.GetCurrentTimeInUnixMillis()
 		klog.V(2).InfoS("creating backend set for ingress", "ingress", klog.KObj(ingress), "backendSetName", bsName)
 		artifact, artifactType := stateStore.GetTLSConfigForBackendSet(bsName)
-		backendSetSslConfig, err := GetSSLConfigForBackendSet(ingress.Namespace, artifactType, artifact, lb, bsName, c.defaultCompartmentId, c.client)
```

Comment:
For cleaner code's sake, maybe instead of passing ingresses namespace and GetIngressExtractCaFromTlsCrt's result everytime, we can pass the entire ingress object and do that deeper in the call stack, but I don't want to mess with the mantainers architecture, but let me know, just a suggestion.